code/modules/clothing/head/helmet.dm

```diff
@@ -399,6 +408,9 @@ GLOBAL_LIST_INIT(allowed_helmet_items, list(
 	camera = new /obj/structure/machinery/camera(src)
 	camera.network = list(CAMERA_NET_OVERWATCH)
 
+	if(length(inserted_visors) || length(built_in_visors))
+		actions_types += /datum/action/item_action/cycle_helmet_huds
+
 	..()
```

Comment:
Init requires a return of parent (also this should probably be called before all the code at 389-413